src/network/networkd-dhcp-common.c

```diff
@@ -95,6 +95,8 @@ void network_adjust_dhcp(Network *network) {
         }
 
         network_adjust_dhcp4(network);
+        if (network->dhcp6_use_dnr < 0)
+                network->dhcp6_use_dnr = network->dhcp6_use_dns;
```

Comment:
please do not change the network object like this. Instead add a helper that resolves this:

```c
static inline network_dhcp6_use_dnr(Network *n) {
        assert(n);
        return network->dhcp6_use_dnr < 0 ? network->dhcp6_use_dns : network->dhcp6_use_dnr;
}
```

Reply:
(and put that somewhere in the header)